### PR TITLE
better grad acc tests

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do?

This PR updates the grad acc tests so that we are able to better catch issues. We check the grad norm also in addition to the loss. We are remove some redundant tests. As promised @vasqu. Just read the new test directly, don't bother checking the old tests

### CODE REVIEW

File: tests/trainer/test_trainer.py [hunk 1]
```diff
@@ -149,314 +148,127 @@ def test_tf32(self):
...
-            )
-            self.assertEqual(
-                trainer.accelerator.gradient_accumulation_steps,
-                1,
-                "Trainer should not leak gradient_accumulation_steps to the Accelerator. ",
-            )
-
-    def test_gradient_accumulation_grad_norm_consistency(self):
+            trainer = Trainer(model=RegressionModel(), args=args, train_dataset=RegressionDataset())
+            self.assertEqual(trainer.accelerator.gradient_accumulation_steps, 1)
```
Comment (vasqu) [inline tests/trainer/test_trainer.py:161]: We can add a reference to the issue imo

File: tests/trainer/test_trainer.py [hunk 2]
```diff
@@ -149,314 +148,127 @@ def test_tf32(self):
...
+        baseline and an equivalent GAS run.
+        """
+        # Looser tolerance: without num_items_in_batch each micro-batch is independently
+        # mean-reduced, so losses won't match as tightly. Uses bs=4/GAS=2 to limit variance.
+        self._check_gradient_accumulation(
+            base_batch_size=8,
+            gas_batch_size=4,
+            gas_steps=2,
+            loss_tolerance=0.1,
+            model_accepts_loss_kwargs=False,
```
Comment (vasqu) [inline tests/trainer/test_trainer.py:253]: Would it make sense with a tighter tolerance when we accept loss kwargs as last test case?

**Review by vasqu — APPROVED** [review]
Nice thank you, just thinking if we should have a 4th case with proper check against different gas bsz x step (4:2 vs 8:1 which is tested for both)

File: tests/trainer/test_trainer.py (hunk 2 above)
Comment (SunMarc) [reply tests/trainer/test_trainer.py:253]: do have a tigher tolerance here already test_gradient_accumulation_grad_norm_with_num_items_in_batch

File: tests/trainer/test_trainer.py (hunk 1 above)
Comment (SunMarc) [reply tests/trainer/test_trainer.py:161]: done
